wear/src/main/java/io/homeassistant/companion/android/conversation/views/ConversationView.kt

```diff
@@ -30,28 +43,81 @@ fun ConversationResultView(
             },
             timeText = { TimeText(visible = !scrollState.isScrollInProgress) }
         ) {
-            ThemeLazyColumn(
-                state = scrollState
+            ScalingLazyColumn(
+                state = scrollState,
+                horizontalAlignment = Alignment.Start
             ) {
                 item {
-                    Text(
-                        text = conversationViewModel.speechResult.ifEmpty {
-                            if (conversationViewModel.supportsConversation)
-                                stringResource(R.string.no_results)
-                            else
-                                stringResource(R.string.no_conversation_support)
-                        },
-                        modifier = Modifier.padding(40.dp)
-                    )
+                    Column {
+                        Spacer(Modifier.padding(24.dp))
+                        SpeechBubble(
+                            text = conversationViewModel.speechResult.ifEmpty {
+                                if (conversationViewModel.supportsConversation)
+                                    stringResource(R.string.no_results)
+                                else
+                                    stringResource(R.string.no_conversation_support)
+                            },
+                            false
+                        )
+                        Spacer(Modifier.padding(8.dp))
+                    }
                 }
                 if (conversationViewModel.conversationResult.isNotEmpty())
                     item {
-                        Text(
-                            text = conversationViewModel.conversationResult,
-                            modifier = Modifier.padding(top = 8.dp, start = 32.dp)
-                        )
+                        Row {
+                            if (conversationViewModel.isHapticEnabled.value)
+                                LocalHapticFeedback.current.performHapticFeedback(HapticFeedbackType.LongPress)

```

Comment:
Haptic feedback will be performed on every recomposition while there is a result. This should use a side-effect instead so it is executed only once.

Reply:
Looks like a `SideEffect` will actually run on every recomposition as well so I guess a `LaunchedEffect` is more proper to handle that? I didnt think this page would recompose again after displaying the text since we don't do anything else and the activity closes when the screen turns off